weave/trace/vals.py

```diff
@@ -481,6 +481,8 @@ def make_trace_obj(
     box_val = box.box(val)
     if isinstance(box_val, pydantic_v1.BaseModel):
         box_val.__dict__["ref"] = new_ref
+    elif box_val is None or isinstance(box_val, bool):
```

Comment:
please add a larger comment here as to why we have this exception and what the implications are